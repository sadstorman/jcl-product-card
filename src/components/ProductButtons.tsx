import { ProductContext } from './ProductCard';
import React, { CSSProperties, useCallback, useContext } from 'react';
import styles from "../styles/styles.module.css";

export interface Props {
    className?: string;
    style?: CSSProperties
}

export const ProductButtons = ({ className, style }: Props) => {

    // Extraer mas count del context
    const { counter, increaseBy, maxCount } = useContext(ProductContext)

    // isMaxReached = useCallback, dependencias [counter, maxCount] 
    // TRUE si el count === maxCount
    // FaLSE si es distinto
    const isMaxReached = useCallback(
        () => !!maxCount && counter === maxCount,
        [counter, maxCount]
    )
    

    return (
        <div className={`${styles.buttonsContainer} ${className} `}
            style={style}
        >
            <button className={styles.buttonMinus} onClick={() => { increaseBy(-1) }} >
                -
            </button>

            <div className={styles.countLabel} >
                {counter}
            </div>

            <button className={`${styles.buttonAdd} ${(isMaxReached()) ? styles.disabled : ''}`} onClick={() => { increaseBy(+1) }}>
                +
            </button>
        </div>
    )
}