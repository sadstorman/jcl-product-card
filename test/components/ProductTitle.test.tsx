import React from 'react';
import renderer from 'react-test-renderer'
import { ProductTitle, ProductCard } from '../../src/components';
import { producto1 } from './data/products';


describe('ProductTitle', () => {

    test('debe de mostrar el comp correctamente con el titulo personalizado', () => {
        const wrapper = renderer.create(
            <ProductTitle title='custom product' className='custom-class' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

    test('debe de mostrar el componente con el nombre del producto', () => {

        const wrapper = renderer.create(
            <ProductCard product={producto1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

});
