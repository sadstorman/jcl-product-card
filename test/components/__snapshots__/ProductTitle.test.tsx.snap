// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el comp correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription custom-class "
>
   
  custom product
   
</span>
`;

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined "
  >
     
    Coffe Mug - Card - sin imagen
     
  </span>
</div>
`;
